packages/message-pickup-repository-pg/src/PostgresMessagePickupRepository.ts

```diff
@@ -0,0 +1,572 @@
+
+    try {
+      // Add a listener to the specified Pub/Sub channel
+      await this.pubSubInstance?.addChannel(channel, async (connectionId: string) => {
+        this.logger?.debug(
+          `[initializeMessageListener] Received new message on channel: ${channel} for connectionId: ${connectionId}`,
+        )
+
+        // Fetch the local live session for the given connectionId
+        const pickupLiveSession = await this.findLocalLiveSession(connectionId)
+
+        if (pickupLiveSession) {
+          this.logger?.debug(
+            `[initializeMessageListener] ${this.instanceName} found a LiveSession on channel: ${channel} for connectionId: ${connectionId}. Delivering messages.`,
+          )
+
+          // Deliver messages from the queue for the live session
+          await this.agent?.messagePickup.deliverMessagesFromQueue({
+            pickupSessionId: pickupLiveSession.id,
+          })
+        } else {
+          this.logger?.debug(
+            `[initializeMessageListener] No LiveSession found on channel: ${channel} for connectionId: ${connectionId}.`,
+          )
+        }
+      })
+
+      this.logger?.info(`[initializeMessageListener] Listener successfully added for channel: ${channel}`)
+    } catch (error) {
+      this.logger?.error(`[initializeMessageListener] Error initializing listener for channel ${channel}: ${error}`)
+      throw new Error(`Failed to initialize listener for channel ${channel}: ${error}`)
+    }
+  }
+
+  /**
+   * This method allow create database and tables they are used for the operation of the messageRepository
+   *
+   */
+  private async buildPgDatabase(): Promise<void> {
+    this.logger?.info(`[buildPgDatabase] PostgresDbService Initializing`)
+
+    const clientConfig = {
+      user: this.postgresUser,
+      host: this.postgresHost,
+      password: this.postgresPassword,
+      port: 5432,
+    }
+
+    const poolConfig = {
+      ...clientConfig,
+      database: this.postgresDatabaseName,
+    }
+
+    const client = new Client(clientConfig)
+
+    try {
+      await client.connect()
+
+      // Check if the database already exists.
+      const result = await client.query('SELECT 1 FROM pg_database WHERE datname = $1', [this.postgresDatabaseName])
+      this.logger?.debug(`[buildPgDatabase] PostgresDbService exist ${result.rowCount}`)
+
+      if (result.rowCount === 0) {
+        // If it doesn't exist, create the database.
+        await client.query(`CREATE DATABASE ${this.postgresDatabaseName}`)
+        this.logger?.info(`[buildPgDatabase] PostgresDbService Database "${this.postgresDatabaseName}" created.`)
+      }
+
+      // Create a new client connected to the specific database.
+      const dbClient = new Client(poolConfig)
+
+      try {
+        await dbClient.connect()
+
+        // Check if the 'messagesTableName' table exists.
+        const messageTableResult = await dbClient.query(`SELECT to_regclass('${messagesTableName}')`)
```

Comment:
What about create table if not exists statements? With multi instance this could fail i guess due to multiple processes doing async query + create